### ISSUE TITLE
Running GPT2 finetuned gives completely wrong output compared to running Pytorch

### ISSUE THREAD

**Issue by hoaileba** [NVIDIA/TensorRT#3354]
## Description

I have a finetuned GPT2 model for a specialized task. I have converted to ONNX and TensorRT. However, when running, the text generated from the converted model is completely different from when I run it with plain Pytorch. (I do exactly as the process in the demo in the repository)

But when I use the original model from HuggingFace directly, the results of both models (converted and pytorch) are exactly the same.

Please help me fix this error

this from converted model 

![image](https://github.com/NVIDIA/TensorRT/assets/56421735/bd1e37df-1d99-4fbe-bd03-f23404d51262)

this from model pytorch (HuggingFace)

![image](https://github.com/NVIDIA/TensorRT/assets/56421735/437ccae1-a494-4998-968e-f018b12ed74e)

## Environment

**TensorRT Version**:8.6.1

**NVIDIA GPU**: RTX 3090

**NVIDIA Driver Version**: 520.61.05

**CUDA Version**: 11.8

**CUDNN Version**: 8

Operating System: ubuntu18.04

Python Version (if applicable):3.8

PyTorch Version (if applicable):1.10.2

**Comment by zerollzeng** [comment]
1. Is the accuracy of onnx model(onnxruntime) consistent with pytorch?
2. Is the accuracy of TRT consistent with onnxruntime?

**Comment by ttyio** [comment]
closing since no activity for more than 3 weeks, pls reopen if you still have question, thanks!
